eox_nelp/openedx_filters/xapi/filters.py

```diff
@@ -361,3 +362,41 @@ def generate_group(group_data):
             group.addmember(agent)
 
         return group
+
+
+class XApiCertificateContextFilter(PipelineStep):
+    """This filter is designed to update the certificate context extensions value.
+
+    How to set:
+        OPEN_EDX_FILTERS_CONFIG = {
+            "event_routing_backends.processors.xapi.transformer.xapi_transformer.get_context": {
+                "pipeline": ["eox_nelp.openedx_filters.xapi.filters.XApiCertificateContextFilter"],
+                "fail_silently": False,
+            },
+        }
+    """
+
+    def run_filter(self, transformer, result):  # pylint: disable=arguments-differ, unused-argument
+        """This adds jws-certificate-location extension
+
+        Arguments:
+            transformer <XApiTransformer>: Transformer instance.
+            result <Context>: Context related to an event.
+
+        Returns:
+            Context: Modified context.
+        """
+        certificate_id = transformer.get_data("data.certificate_id")
+
+        if certificate_id:
+            certificate_url = transformer.get_object_iri("certificates", certificate_id)
+            extension = {"http://id.tincanapi.com/extension/jws-certificate-location": certificate_url}
+
+            if result.extensions:
```

Comment:
The result has been also defined with the extension attribute?
I mean we can ensure there would not fall in an AtrributeError

Reply:
Result is a [Context](https://github.com/RusticiSoftware/TinCanPython/blob/3.x/tincan/context.py#L34) instance and its   definition class already includes extensions attribute